src/shared/menu.ts

```diff
@@ -317,22 +326,32 @@ export function makeMenuDropdown(
  * @param title The text to be displayed for this item
  * @param command The command to be executed when this item is clicked
  * @param key A unique identifier used for identifying the command to be executed on click
+ * @param active A function to determine whether this item should be rendered as "active"
+ * @param cssClasses Additional css classes to be applied to this dropdown item
  */
 export function dropdownItem(
     title: string,
     command: MenuCommand,
-    key: string
+    key: string,
+    active?: (state: EditorState) => boolean,
+    cssClasses?: string[]
 ): MenuCommandEntry {
     const button = document.createElement("button");
     button.type = "button";
-    button.className = `s-btn s-btn__unset grid--cell ta-left px12 py4 h:bg-black-050 c-pointer js-editor-btn`;
     button.dataset.key = key;
     button.innerHTML = title;
```

Comment:
Apologies for the unsolicited feedback from a total stranger and admittedly I know very little about the rest of the infrastructure around this particular piece of code but...

If `title` is just a string is using `.innerHTML` (a known XSS sink) worth the risk here? Perhaps `.innerText` would be safer option?

Reply:
No need to apologize. Great that you're looking out!

You're right, `title` as it is right now is only used as a string and we should switch to `innerText` instead. We've got `cssClasses` as a secondary mechanism to provide additional styling if we need to.

Personally, I don't see the risk of running into an XSS here. We're not passing user-generated data to the `title` parameter and are in full control what we're passing as part of the editor. The only way someone could exploit this is by sneaking a vulnerability into this codebase - which of course can happen, but `innerHTML` is the least of our concerns in that case :)